packages/core/src/common/keybinding.ts

```diff
@@ -115,32 +208,45 @@ export class KeybindingRegistry {
 
     registerKeybindings(...bindings: Keybinding[]): void {
         for (const binding of bindings) {
+            // this.keybindingIndex.setDefaultKeybinding(binding);
             this.registerKeyBinding(binding);
         }
     }
 
     /**
-     * Adds a keybinding to the registry.
+     * Adds a default keybinding to the registry.
      *
      * @param binding
      */
     registerKeyBinding(binding: Keybinding) {
-        const existing = this.keybindings[binding.keyCode.keystroke];
-        if (existing) {
-            const collided = existing.filter(b => b.context === binding.context);
+        const existingBindings = this.keybindingIndex.getKeybindingsForKeystroke(binding.keyCode.keystroke);
+        if (existingBindings) {
+            const collided = existingBindings.filter(b => b.context === binding.context);
             if (collided.length > 0) {
                 this.logger.warn(`Collided keybinding is ignored; `, binding, ' collided with ', collided.join(', '));
                 return;
             }
         }
-        const { keyCode, commandId } = binding;
-        const bindings = this.keybindings[keyCode.keystroke] || [];
-        bindings.push(binding);
-        this.keybindings[keyCode.keystroke] = bindings;
 
-        const commands = this.commands[commandId] || [];
-        commands.push(binding);
-        this.commands[commandId] = bindings;
+        this.keybindingIndex.setDefaultKeybinding(binding);
+    }
+
+    /**
+     * Adds a keybinding to the registry.
+     *
+     * @param binding
+     */
+    setKeybinding(binding: Keybinding) {
```

Comment:
Can we call the method `add` or `register` or something like that: just as you did for the documentation.

Reply:
I was thinking maybe register/override (register for contributions) and overriding when coming from the config file?